app/src/lib/app-state.ts

```diff
@@ -808,21 +808,24 @@ export interface ICherryPickState {
   readonly branchCreated: boolean
 }
 
-/** State associated with a cherry pick being performed on a repository */
-export interface ISquashState {
+/**
+ * Undo state associated with a multi commit operation being performed on a
+ * repository.
+ */
+export interface IMultiCommitOperationUndoState {
```

Comment:
Not super happy with the name since it's only used for squash and reordering, but maybe it can be used for others? I haven't spent much time looking into that yet 😳

Reply:
Yeah. It should be able to be used for cherry-picking when it is refactored to multi-commit-operation. I like it!